### PR TITLE
[tune/doc] improve `log_to_file` doc.

### PR DESCRIPTION
**Pull request by xwjiang2010**
improve `log_to_file` doc.
- Updated description
- Added a section about where to find the logged output
- Added a section about how to leave feedback on this feature
- Added caveats

Signed-off-by: xwjiang2010 <xwjiang2010@gmail.com>

## Related issue number
#31902

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/tune/tutorials/tune-output.rst [hunk 1]
```diff
@@ -149,8 +149,28 @@ respectively:
...
+Where to find ``log_to_file`` files?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+If your Tune workload is configured with syncing to head node, then the corresponding ``log_to_file`` outputs
+can be located under each trial folder.
+If your Tune workload is instead configured with syncing to cloud, then the corresponding ``log_to_file``
+outputs are *NOT* synced to cloud and can only be found in the worker nodes that the corresponding trial happens.
+
+.. note::
+    This can get ugly when the trainable is moved across different nodes throughout its lifetime like in some
+    scheduler's case. We may prioritize enabling this if there are enough user requests.
```
Comment (Yard1) [inline doc/source/tune/tutorials/tune-output.rst:166]:
```suggestion
    This can get ugly when the trainable is moved across different nodes throughout its lifetime. This can happen with some schedulers. We may prioritize enabling this if there are enough user requests. Please open an issue on Ray GitHub if this impacts you.
```

**Review by Yard1 — APPROVED** [review]

File: doc/source/tune/tutorials/tune-output.rst [hunk 2]
```diff
@@ -149,8 +149,30 @@ respectively:
...
+    This can happen with some schedulers or with node failures.
+    We may prioritize enabling this if there are enough user requests.
+    Please open an issue on Ray GitHub if this impacts you.
+
+
+Leave us feedback on this feature
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+We know that logging and observability can be a huge performance boost for your workflow. Let us know what is your
+ideal way of interacting with logging that happens in trainables. To leave your thoughts, please go to
+[Ray GitHub](https://github.com/ray-project/ray/issues)
```
Comment (richardliaw) [inline doc/source/tune/tutorials/tune-output.rst:175]: can you create a concrete issue that we can direct people to?

**Review by richardliaw — APPROVED** [review]

File: doc/source/tune/tutorials/tune-output.rst (hunk 2 above)
Comment (xwjiang2010) [reply doc/source/tune/tutorials/tune-output.rst]: linked an issue for people to file to.

**Review by krfricke — APPROVED** [review]
LGTM can you summarize the changes in the PR description?

File: doc/source/tune/tutorials/tune-output.rst [hunk 3]
```diff
@@ -165,14 +165,15 @@ outputs are *NOT* synced to cloud and can only be found in the worker nodes that
...
+    If this impacts your workflow, consider commenting on
+    [this ticket](https://github.com/ray-project/ray/issues/32142).
 
 
 Leave us feedback on this feature
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 We know that logging and observability can be a huge performance boost for your workflow. Let us know what is your
-ideal way of interacting with logging that happens in trainables. To leave your thoughts, please go to
-[Ray GitHub](https://github.com/ray-project/ray/issues)
+ideal way of interacting with logging that happens in trainables. To leave your thoughts, please comment on
```
Comment (angelinalg) [inline doc/source/tune/tutorials/tune-output.rst:175]:
```suggestion
preferred way to interact with logging in trainables. Leave your comments on
```
